src/MathCwfs.cc

```
@@ -0,0 +1,1156 @@
+                  450 * e2 - 225) *
+                     x6 +
+                 (180 * e14 + 225 * e12 * y2 + 360 * e12 + 450 * e10 * y2 +
+                  540 * e10 + 675 * e8 * y2 + 720 * e8 + 900 * e6 * y2 +
+                  720 * e6 + 675 * e4 * y2 + 540 * e4 + 450 * e2 * y2 +
+                  360 * e2 + 225 * y2 + 180) *
+                     x4 +
+                 (-36 * e16 + 360 * e14 * y2 - 72 * e14 - 1275 * e12 * y4 +
+                  720 * e12 * y2 - 108 * e12 - 2550 * e10 * y4 +
+                  1080 * e10 * y2 - 144 * e10 - 3825 * e8 * y4 +
+                  1440 * e8 * y2 - 180 * e8 - 5100 * e6 * y4 + 1440 * e6 * y2 -
+                  144 * e6 - 3825 * e4 * y4 + 1080 * e4 * y2 - 108 * e4 -
+                  2550 * e2 * y4 + 720 * e2 * y2 - 72 * e2 - 1275 * y4 +
+                  360 * y2 - 36) *
+                     x2 -
+                 36 * e16 * y2 + 180 * e14 * y4 - 72 * e14 * y2 -
+                 125 * e12 * y6 + 360 * e12 * y4 - 108 * e12 * y2 -
+                 250 * e10 * y6 + 540 * e10 * y4 - 144 * e10 * y2 -
+                 375 * e8 * y6 + 720 * e8 * y4 - 180 * e8 * y2 - 500 * e6 * y6 +
+                 720 * e6 * y4 - 144 * e6 * y2 - 375 * e4 * y6 + 540 * e4 * y4 -
+                 108 * e4 * y2 - 250 * e2 * y6 + 360 * e2 * y4 - 72 * e2 * y2 -
+                 125 * y6 + 180 * y4 - 36 * y2) *
+                num2_9 / den2_9;
+
+            temp += pow(Z[19], 2) * (12) * (-400) * pow(r2, 3) / den2_10;
+            temp += pow(Z[20], 2) * (12) * (-400) * pow(r2, 3) / den2_10;
+
+            temp += pow(Z[21], 2) * (7) * 576 *
+                    ((e4 - 5 * e2 * x2 - 5 * e2 * y2 + 3 * e2 + 5 * x4 +
+                      10 * x2 * y2 - 5 * x2 + 5 * y4 - 5 * y2 + 1) *
+                     (e4 - 15 * e2 * x2 - 15 * e2 * y2 + 3 * e2 + 25 * x4 +
+                      50 * x2 * y2 - 15 * x2 + 25 * y4 - 15 * y2 + 1)) /
+                    den2_11;
+
+            out[ii] = temp;
+        }
+    }
+    return result;
```

Please raise an exception if `atype` is neither "1st" nor "2nd".

Please also consider changing `atype` to an enum class (though that is a breaking change and I don't know how widely this code is used). The argument is that it makes the API a bit clearer (the options are self-documenting) and catches errors a bit earlier.

---

Done in the commit: 6d31edc. I am considering to put the `enum` in the Python level. But I am not sure. It will be another JIRA ticket after I have a decision. Thanks!

---

pybind11 does a fine job wrapping C++ enum classes, and I think you will have to do that if you want the new API to be usable from Python.